src/components/OntologyExplorer/index.tsx

```diff
@@ -131,22 +136,65 @@ export default function OntologyExplorer({ graph }: OntologyExplorerProps): JSX.
 
     Side effect: sets the render & simulation state.
     */
+
     if (dagState) {
-      const { nodes, links } = dagState;
+      const { nodes, links } = dagState;      
+
+      const nodeToHullRoot = new Map();
+      let flag = true;
+      let height = 7;
+      const allHullRoots: string[] = [];
+      while (flag && height >= 2) {
+        flag=false;
+        const hullRoots = [...heightMap].filter(([_, v]) => v === height ).map(([k,_])=>k) // get all hullRoots of a certain height
```

Comment:
Nit:  when the map is large, this is an inefficient way to perform this operation (creates and destroys a lot of objects for what is essentially a simple loop/test).  For small collections it is fine, but if the ontologies (ie, size of heightMap) gets large, it is going to be expensive.

The alternative is the (more verbose) for-of loop (or Map.forEach method if you prefer).